bambi/models.py

```diff
@@ -285,9 +292,15 @@ def add_formula(self, fixed=None, random=None, priors=None,
                 data[cats] = data[cats].apply(lambda x: x.astype('category'))
 
             if '~' in fixed:
+                event = re.match(r'^((\S+)\[(\S+)\])\s*~(.*)$', fixed)
+                if event is not None:
+                    fixed = '{}~{}'.format(event.group(2),event.group(4))
                 y, X = dmatrices(fixed, data=data)
                 y_label = y.design_info.term_names[0]
-                self.add_y(y_label, family=family, link=link)
+                self.add_y(y_label, family=family, link=link,
+                    data=None if event is None else
```

Comment:
Maybe put the data assignment on its own line, for clarity. And the data dict inside `pd.DataFrame` could also be broken up to make things clearer.